teps/0053-nested-triggers.md

```diff
@@ -201,4 +202,86 @@ spec:
 Notice how simple it is for the Trigger author to write the logic around their trigger
 when the common logic is implemented at the top level. Trigger authors can also manage
 their own set of downstream triggers as you can specify additional triggers alongside
-a binding and template once the interceptor chain completes.
\ No newline at end of file
+a binding and template once the interceptor chain completes.
+
+## Alternatives
+
+There are three alternatives that have been identified to provide some of the same benefits.
+
+### Option 1: Select Downstream Triggers With Labels
+
+Add a `triggerSelector` field that allows a trigger to target downstream triggers based off a label selector:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: top-level-trigger
+spec:
+  interceptors:
+    - cel:
+        filter: >-
+          header.match('header-filter', 'a-header-filter')
+        - key: example-overlay
+          expression: body.example.value
+  triggerSelector:
+    matchLabels:
+      key: value
+```
+
+### Option 2: Interceptor Groups
+
+Specify the interceptor groups at the top level and process each group on trigger:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: my-el
+spec:
+  groups: 
+  - name: github-group
+    interceptors:
+      - my-interceptor
+      - add-extra-body
+    triggerSelector:
+      matchLabels:
+        app=foo
+        anotherval=xyz
+  - name: another-group
+    interceptors:
+      - my-interceptor
+    triggerSelector:
+       matchLabels:
+         app=foo
+```
+
+This can work well with running a root EventListener interceptor with a `labelSelector`. One challenge here is that these `InterceptorGroups` become another separate API resource that would need to be managed and potentially added to the API. This could be added onto the existing Trigger resource, but then it becomes closer to Option 1.
```

Comment:
I'm not sure if we necessarily need a separate InterceptorGroups CRD. In the original proposal, the chain of interceptors was embedded in a top level Trigger with that top level Trigger then pointing to other Triggers. Here, the chain of interceptors is embedded in the EventListener and that EventListener then calls a bunch of Triggers.

So, the main difference is that instead of an additional top level trigger, the information is embedded inside the EL itself.

We could even get rid of the `groups` portion and simply add an optional interceptor chain to each EventListenerTrigger or something.

Reply:
> This could be added onto the existing Trigger resource, but then it becomes closer to Option 1.

My $0.02 here is that it makes a bit more sense to add it to EL instead of the Trigger resource. It keeps the Trigger spec simple for the end user. Also, keeping it in the EL means that there is a single resource to manage (the EL) instead of an EL as well as the top level Trigger.

Reply:
I think and advantage of this approach is readability - we have the list of top interceptors in one place, so when adding a trigger I can check in there, and decide which labels to add to it so that I end up under the correct top trigger.

Reply:
when we think about the use cases here, interceptor groups are providing a capability to centralize the processing of a chain of interceptors. i have come around a bit on whether that would end up needing to be its own CRD. This fits nicely into the EventListener object, especially when you think about the labelSelector feature just merged in tektoncd/triggers#970. That data is stored _only_ in the eventlistener, so it would make sense that `groups` (with their own label selectors) would only be exposed in the EventListener resource.